supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemJobExecutionActor.scala

```diff
@@ -41,14 +43,24 @@ class SharedFileSystemJobExecutionActor(override val jobDescriptor: BackendJobDe
     case abortResponse: AbortedResponse =>
       context.parent ! abortResponse
       context.stop(self)
+    case KvPair(key, id@Some(jobId)) if key.key == SharedFileSystemJob.JobIdKey =>
+      // Successful operation ID lookup during recover.
+      executor ! Recover(SharedFileSystemJob(jobId))
+    case KvKeyLookupFailed(_) =>
+      // Missed operation ID lookup during recover, fall back to execute.
+      executor ! Execute
+    case KvFailure(_, e) =>
+      // Failed operation ID lookup during recover, crash and let the supervisor deal with it.
+      completionPromise.tryFailure(e)
+      throw new RuntimeException("Failure attempting to look up JES operation ID for key " + jobDescriptor.key, e)
```

Comment:
TOL might it be worth factoring this out to a common trait if recovery handling is generally going to entail c/p of this block?